src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1000,7 +1000,9 @@ private void AddOrReplacePeerTip(int networkPeerId, uint256 newTip)
             if (oldTipHash != null)
             {
                 ChainedHeader oldTip = this.chainedHeadersByHash.TryGet(oldTipHash);
-                this.RemovePeerClaim(networkPeerId, oldTip); // TODO: do we need to throw if oldTip is null (it is expected to be in the dictionary if its not its a potential bug)
+
+                if (oldTip != null)
```

Comment:
No as in 
![dkur9tk](https://user-images.githubusercontent.com/30052221/43593300-52951a0e-966f-11e8-864f-edb04a16eb05.png)